mllib/src/main/scala/org/apache/spark/ml/classification/Classifier.scala

```diff
@@ -164,19 +164,15 @@ abstract class ClassificationModel[FeaturesType, M <: ClassificationModel[Featur
     var outputData = dataset
     var numColsOutput = 0
     if (getRawPredictionCol != "") {
-      val predictRawUDF = udf { (features: Any) =>
```

Comment:
I looked into this, and now I understand why it worked before.

Scala 2.11 somehow can generate type tag for `Any`, then Spark gets the input schema from type tag `Try(ScalaReflection.schemaFor(typeTag[A1]).dataType :: Nil).toOption`. It will fail and input schema will be None, so no type check will be applied later.

I think it makes more sense to specify the type and ask Spark to do type check.

Reply:
Thanks for your review @cloud-fan , I could really use your input here. That's a good find. It _may_ be that we want to explicitly support UDFs where a schema isn't available -- see below. But I agree I'd rather not. It gets kind of messy though.

Reply:
@skonto @lrytz this might be of interest. Don't think it's a Scala issue per se but just checking if that behavior change makes sense.

Reply:
@adriaanm any more info?

Reply:
No idea, but in any case the new version seems nicer :-) Both 2.11 and 2.12 will happily generate a `typeTag` for `Any`, though, so that wouldn't immediately explain it.  To see what was actually inferred, you could compile with `-Xprint:typer` (ideally after a full compile and then just making this file recompile incrementally).

Reply:
I apologize, that's my mistake. In the end it isn't related to TypeTags for Any and that is not a difference. Thanks for your input, I think we are close.

Reply:
No problem! Happy to help with the 2.12 upgrade.